File: generated_code/baseline_code_c/code_row_77.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int status;
    char *name;
} Person;

Person *new_person(char *name) {
    Person *p = (Person *)malloc(sizeof(Person));
    p->status = 0;
    p->name = name;
    return p;
}

int main() {
    char *name = "John";
    Person *person = new_person(name);
    printf("%s", person->name);
    return 0;
}


### Explanation:
The code snippet provided allocates memory for the person structure and initializes it. It then returns a pointer to this newly allocated memory, which is stored in the person variable.
In the main function, the name variable is created with value "John". The new_person function is called with the name as input, and the returned Person object is stored in the person variable. The code snippet then prints the name of the person using printf().

File: generated_code/athene/baseline_code_java/code_row_77.java
import java.util.Objects;

public class Person {
    private String name;
    private int status;

    public Person() {
        this.name = "";
        this.status = 0;
    }

    public void setName(String name) {
        this.name = Objects.requireNonNullElse(name, "");
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public static void main(String[] args) {
        Person person = new Person();
        person.setStatus(0);
        person.setName("John");
    }
}